perun-core/src/main/java/cz/metacentrum/perun/core/impl/AuthzResolverImpl.java

```diff
@@ -785,7 +807,92 @@ public static PerunPolicy getPerunPolicy(String policyName) throws PolicyNotExis
 		return new ArrayList<>(perunPoliciesContainer.getAllPolicies());
 	}
 
+	/**
+	 * Get RoleManagementRules for the role name from the PerunPoliciesContainer
+	 *
+	 * @param roleName for which will be the rules fetched
+	 * @return RoleManagementRules for the role name
+	 * @throws PolicyNotExistsException of there are no rules for the role name
+	 */
 	public static RoleManagementRules getRoleManagementRules(String roleName) throws RoleManagementRulesNotExistsException {
 		return perunPoliciesContainer.getRoleManagementRules(roleName);
 	}
+
+	@Override
+	public Integer getRoleId(String role) {
+		try {
+			return jdbc.queryForInt("SELECT id FROM roles WHERE name=?", role.toLowerCase());
+		} catch (RuntimeException e) {
+			throw new InternalErrorException(e);
+		}
+	}
+
+	@Override
+	public void setRole(PerunSession sess, Map<String, Integer> mappingOfValues) throws RoleAlreadySetException {
+		String query = prepareQueryToSetRole(mappingOfValues);
+
+		try {
+			jdbc.update(query);
+		} catch (DataIntegrityViolationException e) {
+			throw new RoleAlreadySetException("The role was already set.");
+		} catch (RuntimeException e) {
+			throw new InternalErrorException(e);
+		}
+	}
+
+	@Override
+	public void unsetRole(PerunSession sess, Map<String, Integer> mappingOfValues) throws RoleNotSetException {
+		String query = prepareQueryToUnsetRole(mappingOfValues);
+
+		try {
+			if (0 == jdbc.update(query)) {
+				throw new RoleNotSetException("The role already unset.");
+			}
+		} catch (RuntimeException e) {
+			throw new InternalErrorException(e);
+		}
+	}
```

Comment:
Missing tests for these new impl methods.

Reply:
Added